### DISCUSSION TITLE
Codemagic Help problem with No Provisioning Profile was found for your project

### DISCUSSION THREAD

**Discussion by GitYannHub** [codemagic-ci-cd/codemagic-docs#2566] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

Im getting this error building on ios , im using the codemagic.yml:

`workflows:
  react-native-ios:
    name: React Native iOS
    max_build_duration: 120
    instance_type: mac_mini_m1
    integrations:
      app_store_connect: Codemagic API Key
    environment:
      ios_signing:
        distribution_type: app_store
        bundle_identifier: com.japprends.demoyann2 #A Modifier
      vars:
        BUNDLE_ID: "com.japprends.demoyann2" #A modifier
        XCODE_WORKSPACE: "demoyann2.xcworkspace" # #A Modifier
        XCODE_SCHEME: "demoyann2" # #A Modifier
        APP_STORE_APPLE_ID: 2L6W9Q93F8 # A Modifier
    scripts:
      - name: Install npm dependencies
        script: | 
          npm install
      - name: Run Expo Prebuild
        script: | 
          npx expo prebuild
      - name: Set Info.plist values
        script: | 
          PLIST=$CM_BUILD_DIR/$XCODE_SCHEME/Info.plist
          PLIST_BUDDY=/usr/libexec/PlistBuddy
          $PLIST_BUDDY -c "Add :ITSAppUsesNonExemptEncryption bool false" $PLIST
      - name: Install CocoaPods dependencies
        script: | 
          cd ios && pod install
      - name: Set up provisioning profiles settings on Xcode project
        script: xcode-project use-profiles
      - name: Increment build number
        script: | 
          cd $CM_BUILD_DIR/ios
          LATEST_BUILD_NUMBER=$(app-store-connect get-latest-app-store-build-number "$APP_STORE_APPLE_ID")
          agvtool new-version -all $(($LATEST_BUILD_NUMBER + 1))
      - name: Build ipa for distribution
        script: | 
          xcode-project build-ipa \
            --workspace "$CM_BUILD_DIR/ios/$XCODE_WORKSPACE" \
            --scheme "$XCODE_SCHEME" \
            --archive-flags="-destination 'generic/platform=iOS'"
    artifacts:
      - build/ios/ipa/*.ipa
      - /tmp/xcodebuild_logs/*.log
      - $HOME/Library/Developer/Xcode/DerivedData/**/Build/**/*.app
      - $HOME/Library/Developer/Xcode/DerivedData/**/Build/**/*.dSYM
    publishing:
      email:
        recipients:
          - yann@gmail.com
        notify:
          success: true
          failure: false
      app_store_connect:
        auth: integration

        # Configuration related to TestFlight (optional)
        # Note: This action is performed during post-processing.
        submit_to_testflight: true
        beta_groups: # Specify the names of beta tester groups that will get access to the build once it has passed beta review.
          - group name 1
          - group name 2

        # Configuration related to App Store (optional)
        # Note: This action is performed during post-processing.
        submit_to_app_store: false`

### Expected results

I have a provisioning file in codemagic. I am totally lost. Thanks.

### Actual results

Using Xcode 15.3 (15E204a)
Check demoyann2.xcworkspace build settings
Execute "xcodebuild -workspace /Users/builder/clone/ios/demoyann2.xcworkspace -scheme demoyann2 -showBuildSettings"

Archive demoyann2.xcworkspace
Execute "xcodebuild -workspace /Users/builder/clone/ios/demoyann2.xcworkspace -scheme demoyann2 -archivePath build/ios/xcarchive/demoyann2_eoaiv8h3.xcarchive archive COMPILER_INDEX_STORE_ENABLE=NO CODE_SIGN_STYLE=Manual"

❌  /Users/builder/clone/ios/demoyann2.xcodeproj: error: "demoyann2" requires a provisioning profile. Select a provisioning profile in the Signing & Capabilities editor. (in target 'demoyann2' from project 'demoyann2')

### Build id (optional)

665916eedf4860f09db1a835

_No response_

**Comment by himesh-cm** [comment]
The thread is being continued on intercom.

**Comment by dtrdic** [comment, marked as answer]
Note: issue resolved. The iOS bundle identifier inside the app.json file was different than the one specified in codemagic.yaml
